### PR TITLE
docs(webrisk-samples): add new samples (search, threat diff) and refa…

### PR DESCRIPTION
**Pull request by Sita04**
…ctor existing

1. Adding two new samples:
    Search Hashes
    Compute Threat List diff

2. Refactor existing samples and updated tests to match Java standard guide.

### CODE REVIEW

File: webrisk/src/test/java/webrisk/SnippetsIT.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2020 Google LLC
```
Comment (lesv) [inline webrisk/src/test/java/webrisk/SnippetsIT.java:2]: Is this new code?  Then it should be 2022

File: webrisk/src/main/java/webrisk/SubmitUri.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 Google LLC
```
Comment (lesv) [inline webrisk/src/main/java/webrisk/SubmitUri.java:2]: This appears to be new code, so it should be 2022

File: webrisk/src/main/java/webrisk/SearchUri.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 Google LLC
```
Comment (lesv) [inline webrisk/src/main/java/webrisk/SearchUri.java:2]: 2022?

File: webrisk/src/test/java/webrisk/SnippetsIT.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
+        .that(System.getenv(envVarName))
+        .isNotEmpty();
+  }
+
+  @BeforeClass
+  public static void setUp() throws IOException {
+    final PrintStream out = System.out;
+    ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
```
Comment (lesv) [inline webrisk/src/test/java/webrisk/SnippetsIT.java:61]: This seems wrong here.

File: webrisk/src/test/java/webrisk/SnippetsIT.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+  }
+
+  @AfterClass
+  public static void cleanup() throws IOException {
+    final PrintStream out = System.out;
+    ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
+
+    stdOut.close();
+    System.setOut(out);
```
Comment (lesv) [inline webrisk/src/test/java/webrisk/SnippetsIT.java:77]: Seems a bit pointless.

File: webrisk/src/test/java/webrisk/SnippetsIT.java [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+  @Before
+  public void beforeEach() {
+    stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
+  }
+
+  @After
+  public void afterEach() {
+    stdOut = null;
+    System.setOut(null);
```
Comment (lesv) [inline webrisk/src/test/java/webrisk/SnippetsIT.java:89]: Should always be reset to `out` -- Never to `null`.

**Review by lesv — CHANGES_REQUESTED** [review]
PTAL

File: webrisk/src/test/java/webrisk/SnippetsIT.java (hunk 1 above)
Comment (Sita04) [reply webrisk/src/test/java/webrisk/SnippetsIT.java:2]: Not new code. I just renamed the file and made some updates.

File: webrisk/src/main/java/webrisk/SearchUri.java (hunk 3 above)
Comment (Sita04) [reply webrisk/src/main/java/webrisk/SearchUri.java:2]: Ditto. Not new code.

File: webrisk/src/test/java/webrisk/SnippetsIT.java (hunk 4 above)
Comment (Sita04) [reply webrisk/src/test/java/webrisk/SnippetsIT.java]: I removed this code as it was redundant.
But I don't follow your comment here.

**Comment by Sita04** [comment]
@lesv PTAL at the changes. Thanks.

File: webrisk/src/test/java/webrisk/SnippetsIT.java [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+  @Before
+  public void beforeEach() {
+    stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
+  }
+
+  @After
+  public void afterEach() {
+    stdOut = null;
```
Comment (lesv) [inline webrisk/src/test/java/webrisk/SnippetsIT.java:71]: This ideally should be restored `out`, not `null`.

Comment (lesv) [reply webrisk/src/test/java/webrisk/SnippetsIT.java:71]: setting to `null` makes debugging harder.

Comment (lesv) [reply webrisk/src/test/java/webrisk/SnippetsIT.java:71]: This may also mess with later test files.

**Review by lesv — CHANGES_REQUESTED** [review]
It's bad to set `stdOut` to `null`.  You should always restore to a saved version of `System.out`

File: webrisk/src/test/java/webrisk/SnippetsIT.java (hunk 7 above)
Comment (Sita04) [reply webrisk/src/test/java/webrisk/SnippetsIT.java]: Done. I've modified the code. PTAL, thanks :)

File: webrisk/src/test/java/webrisk/SnippetsIT.java [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class SnippetsIT {
+
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static ByteArrayOutputStream originalOut = new ByteArrayOutputStream();
```
Comment (lesv) [inline webrisk/src/test/java/webrisk/SnippetsIT.java:48]: You probably want `System.out` here.

**Review by lesv — CHANGES_REQUESTED** [review]

**Review by lesv — CHANGES_REQUESTED** [review]
PTAL

**Review by lesv — APPROVED** [review]

**Pull request merged**